apps/files/src/components/PublicLinks/PublicLink.vue

```diff
@@ -0,0 +1,97 @@
+<template>
+    <div class="oc-login" uk-height-viewport>
+    <div class="oc-login-card uk-position-center">
+      <h1 class="oc-login-logo" v-translate>
+        ownCloud
+      </h1>
+      <div class="oc-login-card-body" v-if="loading">
+        <h3 class="oc-login-card-title"><translate>Loading public link…</translate></h3>
+        <oc-spinner></oc-spinner>
+      </div>
+      <div class="oc-login-card-body" v-if="errorMessage">
+        <h3 class="oc-login-card-title"><translate>An error occurred while loading the public link</translate></h3>
+        <span>{{ errorMessage }}</span>
+      </div>
+      <div class="oc-login-card-body" v-if="passwordRequired">
+        <h2 class="oc-login-card-title"><translate>This resource is password-protected.</translate></h2>
+
+        <oc-text-input type="password" v-model="password" :placeholder="passwordPlaceholder"></oc-text-input>
+        <br>
```

Comment:
how can I apply the margin? which class shall I use? THX

Reply:
For adding `uk-margin-` and then size (small, medium, large, xlarge or without a size and it will get default) then you can also assign direction (top, right, bottom, left or no direction for giving it to all of them). Work also for removing margin `uk-margin-remove-` and again direction or no direction for all of them.